From 63a49b56af70db77cfcc8a9d336a737300750200 Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Fri, 27 May 2022 10:31:42 +0300
Subject: [PATCH 018/102] misc: s32: change reg address parsing

Previously, the reg address for the nvram was taken from the parent node
siul2 (in the dts). After refactoring the siul2 node, the nvram has its
own reg property.

Issue: ALB-8868

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/misc/s32cc_siul2_nvram.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/s32cc_siul2_nvram.c b/drivers/misc/s32cc_siul2_nvram.c
index aa20c61ded..c324d2f20c 100644
--- a/drivers/misc/s32cc_siul2_nvram.c
+++ b/drivers/misc/s32cc_siul2_nvram.c
@@ -471,7 +471,7 @@ static int siul2_nvram_set_plat(struct udevice *dev)
 {
 	struct siul2_nvram *nvram = dev_get_platdata(dev);
 
-	nvram->base = dev_read_addr_ptr(dev->parent);
+	nvram->base = (void __iomem *)dev_read_addr_ptr(dev);
 	if (!nvram->base)
 		return -EINVAL;
 
-- 
2.17.1

